Sources/Basics/Concurrency/PID.swift

```diff
@@ -0,0 +1,85 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2025 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import Foundation
+
+public protocol PIDFileHandler {
+    var scratchDirectory: AbsolutePath { get set }
+
+    init(scratchDirectory: AbsolutePath)
+
+    func readPID() -> Int32?
+    func deletePIDFile() throws
+    func writePID(pid: Int32) throws
+    func getCurrentPID() -> Int32
+}
+
+public struct PIDFile: PIDFileHandler {
+    public var scratchDirectory: AbsolutePath
+
+    public init(scratchDirectory: AbsolutePath) {
+        self.scratchDirectory = scratchDirectory
+    }
+
+    /// Return the path of the PackageManager.lock.pid file where the PID is located
+    private var lockFilePath: AbsolutePath {
+        self.scratchDirectory.appending(component: "PackageManager.lock.pid")
+    }
+
+    /// Read the pid file
+    public func readPID() -> Int32? {
```

Comment:
suggestion (non-blocking): consider adding `small` tests that would validate all code paths in isolation.